File: packages/callwave_flutter_method_channel/android/src/main/kotlin/com/callwave/flutter/methodchannel/manager/ActiveCallRegistry.kt
package com.callwave.flutter.methodchannel.manager

class ActiveCallRegistry {
    private val activeCalls = LinkedHashSet<String>()

    @Synchronized
    fun tryStart(callId: String): Boolean {
        if (activeCalls.isEmpty() || activeCalls.contains(callId)) {
            activeCalls.add(callId)
            return true
        }
        return false
    }

    @Synchronized
    fun remove(callId: String) {
        activeCalls.remove(callId)
    }

    @Synchronized
    fun getActiveCallIds(): List<String> {
        return activeCalls.toList()
    }
}

File: packages/callwave_flutter_method_channel/ios/Classes/manager/ActiveCallRegistry.swift
import Foundation

final class ActiveCallRegistry {
  private var activeCalls: Set<String> = []

  func tryStart(callId: String) -> Bool {
    if activeCalls.isEmpty || activeCalls.contains(callId) {
      activeCalls.insert(callId)
      return true
    }
    return false
  }

  func remove(callId: String) {
    activeCalls.remove(callId)
  }

  func activeCallIds() -> [String] {
    Array(activeCalls)
  }
}
